### ISSUE TITLE
Shouldn't allow to configure without any conf module, or multiple conf modules

### ISSUE THREAD

**Issue by yurivict** [ibus/ibus#1994]
With these arguments ```--disable-gconf --disable-dconf --disable-memconf``` configure still succeeds:
```
  Build gconf modules           no (disabled, use --enable-gconf to enable)
  Build memconf modules         no
  Build dconf modules           no
```

With these arguments ```--enable-gconf --enable-dconf --enable-memconf``` configure also still succeeds:
```
  Build gconf modules           yes
  Build memconf modules         yes
  Build dconf modules           yes (enabled, use --disable-dconf to disable)
```

It is meaningless to build without conf modules (where is configuration saved?). It is also meaningless to build with multiple conf modules (it shouldn't be saved in multiple places).

**Comment by fujiwarat** [comment]
I think it's the installation task.
I'm thinking the sub packages. E.g. ibus-dconf, ibus-gconf, ibus-memconf sub packages. And the package manager selects one of the packages.
If only one package is needed in your distro, the package maintainer does not have to build all modules.

**Comment by yurivict** [comment]
Yes, they can be split out. Most users just need one method, though. It's not like people will change them often. So a good default is sufficient for most users.

**Comment by fujiwarat** [comment]
Most users do not specify the configure options for the ibus-config but do the default.
Currently I don't think to modify this configuration. This is needed under a condition.

**Comment by yurivict** [comment]
When I saw that it can build w/out or with multiple conf option selections, I thought this was meaningless.

**Comment by fujiwarat** [comment]
I think the option is valid at present.
E.g. LiveDVD doesn't want to have ibus-dconf|ibus-gconf and asked to have ibus-memconf but desktop asked ibus-dconf.
Currently we're working on migrating ibus-config to GSettings. Once the task is done, we can deprecate ibus-config or disable it by default.
And then I think  this request will be triaged.

**Comment by yurivict** [comment]
Ok.
